cn0326: made to calibrate before every data read and fixed two bugs

adrimbarean opened the pull request:
To work properly, the sensor has to be calibrated before every data
aquisition. The first bug was the wait condition in the scan method where
the program was waiting for convertion to finish only in continuous
convertion mode. The second bus was that the wait condition for the
convertion was wrong. These changes address these issues.

Signed-off-by: adrimbarean <Andrei.Drimbarean@analog.com>

mirceacaprioru commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -177,14 +174,13 @@ uint32_t AD7793_Scan(enMode mode,  uint8_t ui8channel)
 
-	if(mode == CONTINUOUS_CONV)
+	if(mode == CONTINUOUS_CONV) {
 		//DioClr(CS_PORT, CS_PIN);
```
remove commented code

mirceacaprioru commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -177,14 +174,13 @@ uint32_t AD7793_Scan(enMode mode,  uint8_t ui8channel)
-	ui32result =  SPI_Read(ui8reg_adrr,  reg_size[AD7793_REG_DATA]);
+	ui32result = AD7793_ReadRegister(AD7793_REG_DATA);
 
 	//DioSet(CS_PORT, CS_PIN);
```
same thing

mirceacaprioru commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -241,7 +237,7 @@ void AD7793_Calibrate(uint8_t ui8channel, enMode mode)
-	while ((AD7793_ReadRegister(AD7793_REG_STAT)& RDY_BIT) != RDY_BIT);
+	while ((AD7793_ReadRegister(AD7793_REG_STAT)& RDY_BIT) == RDY_BIT);
 
 	//DioSet(CS_PORT, CS_PIN); todo: asses the importance of the gpio functions
```
commented code

adrimbarean commented:
v0->v1:
- correctly separate all changes into different commits so they can be followed
Changes:
- removed commented code
- changed to floating point operations
- corrected wait condition
- fix wrong variable names
- add calibration before every read
- fixed a bug with SPI communication introduced at porting
- changed a dummy write and read operation to a normal read using wrapper function

stefpopa commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -176,7 +176,7 @@ uint32_t AD7793_Scan(enMode mode,  uint8_t ui8channel)
+	while ((AD7793_ReadRegister(AD7793_REG_STAT)& RDY_BIT) == RDY_BIT);
 
 	ui32result = AD7793_ReadRegister(AD7793_REG_DATA);
 
```
there is no need to store the result in a new variable, we can directly return here:
`return AD7793_ReadRegister(AD7793_REG_DATA);`

commodo commented:
General note about your `Signed-off-by` ; you should use your full name, not your username.
So,  
`Signed-off-by: adrimbarean <Andrei.Drimbarean@analog.com>` --> `Signed-off-by: Andrei Drimbarean <Andrei.Drimbarean@analog.com>`

mirceacaprioru commented:
The commits run AStyle and enforce 80 column limit could be squashed since they are basically just  operations for beautifying code.

commodo commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -155,10 +155,7 @@ uint32_t AD7793_Scan(enMode mode,  uint8_t ui8channel)
-	/* Set value (read command + DATA register address) to write in COMM
-	 * register */
-	uint8_t ui8reg_adrr = (AD7793_COMM_READ
-	                       | AD7793_COMM_ADR(AD7793_REG_DATA));
```
this looks incorrect; you are removing the address value here; this will likely not compile;
it would be a good idea to re-order this commit after commit 6210d47cdd335d0458fed05efaadbf3ba1ec3ff5 ( cn0326: replace spi read function with wrapper function  )
but for now, i guess we could leave it like this

stefpopa commented on projects/ADuCM3029_demo_cn0326/src/AD7793.c:
```diff
@@ -224,7 +228,9 @@ void AD7793_Calibrate(uint8_t ui8channel, enMode mode)
+    while ((AD7793_ReadRegister(AD7793_REG_STAT) & RDY_BIT) == RDY_BIT &&
+    		timeout < 10000)
+    	timeout++;
 }
```
It is a bit uncertain about how long it takes to reach the timeout. Maybe we should consider implementing this more like in this example: https://github.com/analogdevicesinc/linux/blob/rpi-4.9.y/drivers/iio/dac/ad5758.c#L254
Also, how long does it actually take for the RDY_BIT to be set? Do we have a worst case scenario? Is this information available in the datasheet?

stefpopa replied:
But, I think for now we can remove the timeout completely and add fix it later with a different PR

adrimbarean commented:
v2->v3:
- beautify code (squash astyle and 80 column commits)
- return directly from function
- remove unnecessary static denominator

adrimbarean commented:
If all is good I will merge this today.
